cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,217 @@
+"""
+Views for the maintenance app.
+"""
+import logging
+from django.db import transaction
+from django.core.urlresolvers import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import View
+
+from edxmako.shortcuts import render_to_response
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from contentstore.management.commands.utils import get_course_versions
+from util.views import require_global_staff
+
+
+log = logging.getLogger(__name__)
+
+MAINTENANCE_COMMANDS = {
```

Comment:
This kind of approach is also used in LMS Support app here https://github.com/edx/edx-platform/blob/a680919e8c9331b21e9a27d2abd7e3985a8b1a0e/lms/djangoapps/support/views/index.py#L11-L11